gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/BackendsMetricStats.java

```diff
@@ -0,0 +1,119 @@
+    private Map<String, BackendClusterMetricStats> statsMap = new ConcurrentHashMap<>();
+    private final ScheduledExecutorService scheduledExecutor = Executors.newSingleThreadScheduledExecutor();
+
+    @Inject
+    public BackendsMetricStats(GatewayBackendManager gatewayBackendManager, MBeanExporter exporter, MonitorConfiguration monitorConfiguration)
+    {
+        this.gatewayBackendManager = gatewayBackendManager;
+        this.exporter = exporter;
+        this.monitorConfiguration = monitorConfiguration;
+    }
+
+    @PostConstruct
+    public void start()
+    {
+        int refreshSeconds = monitorConfiguration.getMetricRefreshSeconds();
+        log.info("Running periodic metric refresh with interval of %d seconds", refreshSeconds);
+        scheduledExecutor.scheduleAtFixedRate(() -> {
+            try {
+                initMetrics();
+            }
+            catch (Exception e) {
+                log.error(e, "Error refreshing backend metrics");
+            }
+        }, 0, refreshSeconds, TimeUnit.SECONDS);
+    }
+
+    @PreDestroy
+    public void stop()
+    {
+        scheduledExecutor.shutdownNow();
+    }
+
+    public void initMetrics()
+    {
+        // Get current backends from DB
+        Set<String> currentBackends = gatewayBackendManager.getAllBackends().stream()
+                .map(ProxyBackendConfiguration::getName)
+                .collect(Collectors.toSet());
+
+        // Unregister metrics for removed backends
+        for (String registeredBackend : statsMap.keySet()) {
+            if (!currentBackends.contains(registeredBackend)) {
+                try {
+                    exporter.unexportWithGeneratedName(BackendClusterMetricStats.class, registeredBackend);
+                    log.info("Unregistered metrics for removed cluster: %s", registeredBackend);
+                    statsMap.remove(registeredBackend);
+                }
+                catch (Exception e) {
+                    log.error(e, "Failed to unregister metrics for cluster: %s", registeredBackend);
+                }
+            }
+        }
+
+        // Register metrics for added backends
+        for (String backend : currentBackends) {
+            if (!statsMap.containsKey(backend)) {
+                registerBackendMetrics(backend);
+            }
+        }
+    }
+
+    public void registerBackendMetrics(String clusterName)
+    {
+        BackendClusterMetricStats stats = new BackendClusterMetricStats(clusterName, gatewayBackendManager);
+
+        if (statsMap.putIfAbsent(clusterName, stats) == null) {  // null means the stats didn't exist previously and was inserted
+            try {
+                exporter.exportWithGeneratedName(stats, BackendClusterMetricStats.class, clusterName);
+                log.info("Registered metrics for cluster: %s", clusterName);
+            }
+            catch (Exception e) {
+                statsMap.remove(clusterName);
```

Comment:
There are some race conditions here where in theory, the map contained by `exporter` could fall out of sync with `statsMap`. Given this is not high-concurrency (we're only calling it every 30 seconds), I would recommend we just wrap the whole `initMetrics` in a synchronized block, and then we can safely assume everything in single-threaded and just use `HashMap` instead of `ConcurrentHashMap`.